### PR TITLE
Adds setup script for key generation, working towards #853

### PR DESCRIPTION
**Pull request by umbernhard**
Installs signify-openbsd and qrencode, creates a keypair and outputs a QR code corresponding to the public key.

### CODE REVIEW

File: setup-signify.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -euo pipefail 
+
+sudo apt -y install signify-openbsd qrencode
+
+# clear out old keys
+rm -f key.pub key.sec
```
Comment (benadida) [inline setup-signify.sh:7]: should this be `/vx/config/key.pub` etc.?

File: setup-signify.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+# Generate the keypair
+signify-openbsd -G -n -p /vx/config/key.pub -s /vx/config/key.sec
+
+# Output the public key for enrollment into another device
+qrencode -t UTF8 -r key.pub -o -
```
Comment (benadida) [inline setup-signify.sh:13]: same here, should this be `/vx/config/key.pub` ?

**Review by benadida — COMMENTED** [review]
wondering about paths

File: setup-signify.sh (hunk 1 above)
Comment (umbernhard) [reply setup-signify.sh:7]: oh whoops, yeah.

File: setup-signify.sh (hunk 2 above)
Comment (umbernhard) [reply setup-signify.sh]: Fixed

File: setup-signify.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -euo pipefail 
+
+sudo apt -y install signify-openbsd qrencode
+
+# clear out old keys
+rm -f /vx/config/key.pub /vx/config/key.sec
+
+# Generate the keypair
+signify-openbsd -G -n -p /vx/config/key.pub -s /vx/config/key.sec
```
Comment (eventualbuddha) [inline setup-signify.sh:10]: I'm unsure what the permissions should be for `/vx/config` to be able to run this without `sudo`.

**Review by eventualbuddha — APPROVED** [review]
Mostly okay, though unsure about a permission issue.

File: setup-signify.sh (hunk 3 above)
Comment (umbernhard) [reply setup-signify.sh:10]: There's no reason we couldn't run with sudo I don't think, @benadida is that right?

Comment (benadida) [reply setup-signify.sh:10]: vx-admin can write to /Vx/config, and that's the user that will generate the keys.

Comment (umbernhard) [reply setup-signify.sh:10]: so no need for sudo then?

File: setup-signify.sh [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
+    rm -f $wd/key.pub $wd/key.sec
+
+    # Generate the keypair
+    signify-openbsd -G -n -p $wd/key.pub -s $wd/key.sec
+
+    # Output the public key for enrollment into another device
+    qrencode -t UTF8 -r $wd/key.pub -o -
+else 
+    echo "Usage: ./setup-signify [directory]"
+    exit;
```
Comment (eventualbuddha) [inline setup-signify.sh:19]:
```suggestion
    exit 1
```

File: setup-signify.sh [hunk 5]
```diff
@@ -1,13 +1,21 @@
 #!/bin/bash
 set -euo pipefail 
 
-sudo apt -y install signify-openbsd qrencode
+if [ $# -eq 1 ]; then
+    wd=$1
```
Comment (eventualbuddha) [inline setup-signify.sh:5]:
```suggestion
  public_path="${1}/key.pub"
  secret_path="${1}/key.sec"
```

Then use `"${public_path}"` and `"${secret_path}"` (with double-quotes) below.

**Review by benadida — APPROVED** [review]

File: setup-signify.sh (hunk 4 above)
Comment (eventualbuddha) [reply setup-signify.sh:19]: Or, just make this an early exit:

```bash
if [ $# -eq 0 ]; then
    echo "Usage: ./setup-signify [directory]"
    exit 1
fi

# put the rest here

**Review by eventualbuddha — APPROVED** [review]

**Pull request merged**